### PR TITLE
docs(supported-environments): simplify updates

### LINKED ISSUE #561: Simplify Supported Environments page
https://docs.camunda.io/docs/reference/supported-environments/

Currently supported environments page goes into every component separately. 
We should consider simplifying this. If there are differences, we should really think about the WHY and try to align as much as possible. This might also lead to changes in supported environments in the end.

### PR DESCRIPTION
**Pull request by akeller**
Based on internal docu and discussion.

This is a draft since we are waiting for details from Optimize.

Closes #561.

### CODE REVIEW

File: docs/reference/supported-environments.md [hunk 1]
```diff
@@ -1,85 +1,68 @@
...
-- Oracle JDK/JRE 11
-- Open JDK/JRE 11 including builds of the following products:
- - Adopt OpenJDK
+### Desktop Modeler
+- Windows 7 / 10
+- Mac OS X 10.11
+- Ubuntu LTS (latest)
 
-### Docker
+## Camunda Cloud Self-Managed
```
Comment (felix-mueller) [inline docs/reference/supported-environments.md:30]: probably we can call it Camunda Platform 8 Self-Managed

File: docs/reference/supported-environments.md [hunk 2]
```diff
@@ -1,85 +1,68 @@
 ---
 id: supported-environments
 title: "Supported environments"
-description: "Let's take a closer look at supported environments alongside Zeebe, Operate, and Tasklist."
+description: "Find out where to run Camunda Cloud components for SaaS and Self-Managed, including Optimize for both Camunda Cloud and Camunda Platform 7."
 ---
 
-## Zeebe
+## Camunda Cloud SaaS & Self-Managed
```
Comment (felix-mueller) [inline docs/reference/supported-environments.md:7]: Camunda Platform 8 SaaS & Self-Managed

File: docs/reference/supported-environments.md [hunk 3]
```diff
@@ -1,85 +1,68 @@
 ---
 id: supported-environments
 title: "Supported environments"
-description: "Let's take a closer look at supported environments alongside Zeebe, Operate, and Tasklist."
+description: "Find out where to run Camunda Cloud components for SaaS and Self-Managed, including Optimize for both Camunda Cloud and Camunda Platform 7."
```
Comment (felix-mueller) [inline docs/reference/supported-environments.md:4]: Camunda Platform 8 and Camunda Platform 7

**Comment by akeller** [comment]
@felix-mueller good catch, my brain is still not in full rename mode! 

Anything else?

**Comment by akeller** [comment]
I thought we were having merge conflict-type issues, but it turns out the build is failing for real reasons!

Some of the older migration update versions for Optimize include references to connecting to Elasticsearch 6.2.0+. Since we are supporting Camunda Platform 7 users, we likely need to have this information as well. Or do we only need to support through 3.4.x, and if so, can I remove the migration update guides for 2.1 through 3.3?

**Comment by felix-mueller** [comment]
I added Optimize 3.3 because officially still support it till end of june 22. 
We should keep the update guides, but I don't think we need to cover all old versions here for now in supported envs. 
do you think we are done with this?

**Comment by akeller** [comment]
Looks like the build action was stuck, let me rerun this, and then we can get it merged!

**Pull request merged**